graph.py

```diff
@@ -57,7 +57,10 @@ def del_edge(self, n1, n2):
         Delete an edge from the Graph. If an edge doesn't exist pass
         """
         try:
-            self.g[n1].remove(n2)
+            for num in range(len(self.g[n1])):
+                if self.g[n1][num][0] == n2:
+                    self.g[n1].remove(self.g[n1][num])
+                    break
```

Comment:
This is overly complex.  Iterating over a range of the length of a list instead of simply iterating over the list is almost never what you really want.  And I don't think it's what you want here.  Why couldn't you try something less convoluted, like this:

```python
for pair in self.g[n1]:
    if pair[0] == n2:
        self.g[n1].remove(pair)
        break
```